File: examples/l1d/shock-tube-he-air-1T/he-air-1T.py
# he-air-1T.py
config.title = 'Shock tube, helium driving 1T air, 2023-01-27'
gm_he = add_gas_model('ideal-helium-gas-model.lua')
gm_air = add_gas_model('air-5sp-1T-gas-model.lua', 'air-5sp-1T-chem.lua')
config.reacting = True

# Define the tube walls.
add_break_point(-3.0, 0.01)
add_break_point(9.0, 0.01)

# Create the gas-path.
left_wall = VelocityEnd(x0=-2.0, vel=0.0)
driver_gas = GasSlug(p=0.800e6, vel=0.0, T=800.0, gmodel_id=gm_he, ncells=1200)
interface = GasInterface(x0=0.0)
driven_gas = GasSlug(p=290.0, vel=0.0, T=293.0, massf={'N2':0.78,'O2':0.22},
                     gmodel_id=gm_air, ncells=1200)
right_wall = VelocityEnd(x0=9.0, vel=0.0)
assemble_gas_path(left_wall, driver_gas, interface, driven_gas, right_wall)

# Set some time-stepping parameters
config.dt_init = 1.0e-8
config.max_time = 3.0e-3
config.max_step = 90000
add_dt_plot(0.0, 20.0e-6, 1.0e-6)
add_history_loc(1.0)
add_history_loc(1.1)
add_history_loc(2.0)
add_history_loc(2.1)
add_history_loc(3.0)
add_history_loc(3.1)
add_history_loc(4.0)
add_history_loc(4.1)
add_history_loc(5.0)
add_history_loc(5.1)
add_history_loc(6.0)
add_history_loc(6.1)
add_history_loc(7.0)
add_history_loc(7.1)
add_history_loc(8.0)
add_history_loc(8.1)

File: examples/eilmer/2D/shock-tube-he-air-1T/he-air-1T.lua
-- he-air-1T.lua
-- PJ 2023-01-30
config.title = "One-dimensional shock tube with helium driving air."
print(config.title)

quad0 = CoonsPatch:new{p00={x=-2.0,y=0}, p10={x=9.0,y=0},
                       p11={x=9.0,y=0.1}, p01={x=-2.0,y=0.1}}
grid0 = StructuredGrid:new{psurface=quad0, niv=2001, njv=3}

nsp, nmodes = setGasModel('he-air-5sp-1T-gas-model.lua')
print("GasModel nsp= ", nsp, " nmodes= ", nmodes)
config.reacting = true
config.reactions_file = 'he-air-5sp-1T-chem.lua'

function tube_gas(x, y, z)
   -- User-defined function for the initial gas state works in physical space.
   -- Half the domain has high pressure and the other half low pressure.
   local massFractions, p, T
   if x < 0.0 then
      -- Fill the left-half of the volume with high-pressure helium.
      massFractions = {He=1.0, N2=0.0, O2=0.0}
      p = 0.8e6; T = 800.0
   else
      -- and the right-half with low-pressure air.
      massFractions = {He=0.0, N2=0.7778, O2=0.2222}
      p = 290.0; T = 293.0
   end
   -- We use the FlowState object to conveniently set all of
   -- the relevant properties.
   return FlowState:new{p=p, velx=0.0, vely=0.0, T=T, massf=massFractions}
end

-- Define a single block for the tube and fill it with gas conditions.
blks = FBArray:new{grid=grid0, nib=8, njb=1, initialState=tube_gas}

config.flux_calculator = "ausmdv"
config.max_time = 3.0e-3  -- seconds
config.max_step = 60000
config.dt_init = 1.0e-8
config.dt_plot = 20.0e-6
config.dt_hist = 1.0e-6
-- TODO history points

